src/main/java/DilemmaDetector/Simulator/SimulatorEngine.java

```diff
@@ -98,14 +101,26 @@ public SimulatorEngine(Model model, CollisionConsequencePredictor consequencePre
                         SIMULATION_TIME = updateSimulationTime(currentTime);
                     }
                 }
-                else if(!factoryWrapper.isPedestrian(actor)){
+                else if (factoryWrapper.isSurrounding(actor)){
+                    collisionWithSurrounding = true;
+                }
+                else if(factoryWrapper.isVehicle(actor)){
                     if (!actor.equals(mainVehicle)) {
-                        collisionNotWithPedestrian = true;
+                        collisionWithVehicle = true;
                     }
                 }
+                else if(factoryWrapper.isObstacle(actor.getEntityName())){
+                    collisionWithObstacle = true;
+                }
             }
+
+            Set<Actor> collidedInMomentWithoutSurroundingAndObstacles =
+                    collidedInMoment.stream().filter(
+                            a -> !factoryWrapper.isObstacleOrSurrounding(a.getEntityName())).
+                            collect(Collectors.toSet());
```

Comment:
Po pierwsze dziwnie to wcięte jest, a po drugie i ważniejsze, po co to w streamie robić, jeżeli możesz te same checki robić w pętli `for` powyżej, wiesz, tej w której przechodzisz po wszystkich obiektach kolizji i sprawdzasz czy są obstacle czy surrounding...